js/src/path.js

```diff
@@ -40,13 +40,16 @@ export interface Part {
 export default class Path {
   _parts: Array<Part>;
 
-  static parse(str: string): Path {
+  /**
+   * Returns `str` parsed as Path if successful, or a null path with an error message if not.
+   */
+  static parse(str: string): [Path | null, string /* error */] {
```

Comment:
The other option would be to simply remove the error message - or make Path contain its own error message - but ugh.